ts-sdk/src/schema/chain.ts

```diff
@@ -61,12 +62,25 @@ export class CosmosAddressEncodeError extends Data.TaggedError("CosmosAddressEnc
   prefix: string
 }> {}
 
+const HRP = S.String.pipe(
+  S.length({
+    min: 1,
+    max: 83,
+  }, {
+    description: "HRP must be between 1 to 83 US-ASCII characters, inclusive"
+  }),
+  S.pattern(/^[\x21-\x7E]+$/, {
+    description: "HRP characters must be within the range [33-126], inclusive"
+  })
+)
+export type HRP = typeof HRP.Type
```

Comment:
See https://en.bitcoin.it/wiki/BIP_0173#Bech32. Intended for future `fast-check` arbitrary generation.